app/Main.hs

```diff
@@ -188,32 +198,69 @@ eventProcessor submit nextEvent = go
             IssueFanout -> submit NodeCommand.Fanout >> go
             InitHead period -> submit (NodeCommand.Init period) >> go
             CommitToHead utxoToCommit -> submit (NodeCommand.Commit utxoToCommit) >> go
-            Bet _txIn _pp -> liftIO (putStrLn "no betting yet") >> go
+            Bet txIn txInLovelace pp -> do
+              state <- ask
+              let unsignedTx = either error id $ buildBetTx txIn txInLovelace state pp
+                  signedTx = mkSignedTx state unsignedTx
+              submit (NodeCommand.newTx signedTx)
+              go
             Claim collateralTxIn cp -> do
               state <- ask
               let unsignedTx = either error id $ buildClaimTx collateralTxIn state cp
-                  signedTx = signTx state.hsUserCredentials.userSkey unsignedTx
+                  signedTx = mkSignedTx state unsignedTx
               submit (NodeCommand.newTx signedTx) >> go
 
+mkSignedTx :: HeadState -> TxBody AlonzoEra -> Tx AlonzoEra
```

Comment:
nit: is there any reason to pull this out? doesn't really seem worth to just traverse the state

Reply:
Yes, on the second thought, it seems redundant.